drift/lib/src/sqlite3/database.dart

```diff
@@ -132,6 +143,9 @@ abstract class Sqlite3Delegate<DB extends CommonDatabase>
     if (_closeUnderlyingWhenClosed) {
       beforeClose(_db);
       _db.dispose();
+      if (_syncPersistence) {
+        await _db.flush();
```

Comment:
To avoid the new API on the database (see my comments on the sqlite3 PR), we could export the environment through `WasmSqlite` instead of the database. Then we could call `flush()` directly on the file system instead of adding new methods to the database that only make sense on some platforms.